contracts/plugins/assets/curve/CurveAppreciatingRTokenFiatCollateral.sol

```diff
@@ -0,0 +1,196 @@
+                    // untested:
+                    //      validated in other plugins, cost to test here is high
+                    assert(low == 0);
+                }
+
+                // Check RToken status
+                try pairedBasketHandler.isReady() returns (bool isReady) {
+                    if (!isReady) {
+                        markStatus(CollateralStatus.IFFY);
+                    } else if (low == 0 || _anyDepeggedInPool() || _anyDepeggedOutsidePool()) {
+                        // If the price is below the default-threshold price, default eventually
+                        // uint192(+/-) is the same as Fix.plus/minus
+                        markStatus(CollateralStatus.IFFY);
+                    } else {
+                        markStatus(CollateralStatus.SOUND);
+                    }
+                } catch {
+                    // prefer NOT to revert on empty data here: an RToken missing the `isReady()`
+                    // function would error out with empty data just like an OOG error.
+                    markStatus(CollateralStatus.IFFY);
+                }
+            } catch (bytes memory errData) {
+                // see: docs/solidity-style.md#Catching-Empty-Data
+                if (errData.length == 0) revert(); // solhint-disable-line reason-string
+                markStatus(CollateralStatus.IFFY);
+            }
+        } catch (bytes memory errData) {
+            // see: docs/solidity-style.md#Catching-Empty-Data
+            if (errData.length == 0) revert(); // solhint-disable-line reason-string
+            markStatus(CollateralStatus.DISABLED);
+        }
+
+        CollateralStatus newStatus = status();
+        if (oldStatus != newStatus) {
+            emit CollateralStatusChanged(oldStatus, newStatus);
+        }
+    }
+
+    /// @dev Not up-only! The RToken can devalue its exchange rate peg
+    /// @return {ref/tok} Quantity of whole reference units per whole collateral tokens
+    function underlyingRefPerTok() public view virtual override returns (uint192) {
+        // {ref/tok} = quantity of the reference unit token in the pool per LP token
+
+        // {lpToken@t=0/lpToken}
+        uint192 virtualPrice = _safeWrap(curvePool.get_virtual_price());
+        // this is missing the fact that the RToken has also appreciated in this time
+
+        // {BU/rTok}
+        uint192 rTokenRate = divuu(rToken.basketsNeeded(), rToken.totalSupply());
+        // not worth the gas to protect against div-by-zero
+
+        // {ref/tok} = {ref/lpToken} = {lpToken@t=0/lpToken} * {1} * 2{ref/lpToken@t=0}
+        return virtualPrice.mul(rTokenRate.sqrt()).mulu(2); // LP token worth twice as much
+    }
+
+    /// @dev Warning: Can revert
+    /// @param index The index of the token: 0, 1, 2, or 3
+    /// @return low {UoA/ref_index}
+    /// @return high {UoA/ref_index}
+    function tokenPrice(uint8 index) public view override returns (uint192 low, uint192 high) {
```

Comment:
Needs a comment to specify that it only works for when the RToken is token0 in the pool.